### PR TITLE
Feature/chronoquark all timestamps

### PR DESCRIPTION
**Pull request by flotothemoon**
Move all timestamp values into single ChronoQuark

### CODE REVIEW

File: radixdlt-java/src/main/java/com/radixdlt/client/atommodel/quarks/ChronoQuark.java [hunk 1]
```diff
@@ -4,37 +4,44 @@
...
+	private Map<String, Long> timestamps = new HashMap<>();
 
 	private ChronoQuark() {
 	}
 
-	public ChronoQuark(String timestampkey, long timestampValue) {
-		this.timestampKey = Objects.requireNonNull(timestampkey);
-		this.timestampValue = timestampValue;
+	public ChronoQuark(String timestampKey, long timestampValue) {
+		this.timestamps.put(Objects.requireNonNull(timestampKey, "timestampKey is required"), timestampValue);
```
Comment (mpsandiford) [inline radixdlt-java/src/main/java/com/radixdlt/client/atommodel/quarks/ChronoQuark.java]: I would prefer to see the prereqs performed prior to the actual code body, something like this:
```
Objects.requireNoNull(timestampKey, "timestampKey is required");
this.timestamps.put(timestampKey, timestampValue);
```
This keeps the prereqs independent of the actual operation.

File: radixdlt-java/src/main/java/com/radixdlt/client/atommodel/quarks/ChronoQuark.java [hunk 2]
```diff
@@ -4,37 +4,44 @@
...
+	}
+
+	public long getTimestamp() {
+		return getTimestamp(Timestamps.DEFAULT);
+	}
+
+	public long getTimestamp(String type) {
+		return this.timestamps.getOrDefault(Objects.requireNonNull(type, "type is required"), 0L);
```
Comment (mpsandiford) [inline radixdlt-java/src/main/java/com/radixdlt/client/atommodel/quarks/ChronoQuark.java]: Again, would like to see prereqs separate from code body.

File: radixdlt-java/src/main/java/com/radixdlt/client/atommodel/quarks/ChronoQuark.java [hunk 3]
```diff
@@ -4,37 +4,44 @@
...
 	}
 
-	public String getTimestampKey() {
-		return this.timestampKey;
+	public void setTimestamp(String type, long timestamp) {
+		this.timestamps.put(Objects.requireNonNull(type, "type is required"), timestamp);
```
Comment (mpsandiford) [inline radixdlt-java/src/main/java/com/radixdlt/client/atommodel/quarks/ChronoQuark.java]: Again, would like to see prereqs separate from code body.

File: radixdlt-java/src/main/java/com/radixdlt/client/atommodel/quarks/ChronoQuark.java [hunk 4]
```diff
@@ -4,37 +4,44 @@
...
 	}
 
-	public ChronoQuark(String timestampkey, long timestampValue) {
-		this.timestampKey = Objects.requireNonNull(timestampkey);
-		this.timestampValue = timestampValue;
+	public ChronoQuark(String timestampKey, long timestampValue) {
+		this.timestamps.put(Objects.requireNonNull(timestampKey, "timestampKey is required"), timestampValue);
+	}
+
+	public long getTimestamp() {
```
Comment (mpsandiford) [inline radixdlt-java/src/main/java/com/radixdlt/client/atommodel/quarks/ChronoQuark.java]: Would it make sense to have a corresponding `setTimetamp(long)` and constructor to mirror this operation?

File: radixdlt-java/src/main/java/com/radixdlt/client/atommodel/timestamp/TimestampParticle.java [hunk 5]
```diff
@@ -26,7 +26,6 @@ public TimestampParticle(long timestamp) {
 	}
 
 	public Long getTimestamp() {
```
Comment (mpsandiford) [inline radixdlt-java/src/main/java/com/radixdlt/client/atommodel/timestamp/TimestampParticle.java]: Appreciate this didn't change, but I think this should be a `long`, not a wrapped `Long`.

File: radixdlt-java/src/main/java/com/radixdlt/client/atommodel/quarks/ChronoQuark.java (hunk 3 above)
Comment (mpsandiford) [reply radixdlt-java/src/main/java/com/radixdlt/client/atommodel/quarks/ChronoQuark.java]: Actually, can we get rid of this method by providing a richer set of constructors?

This is the only mutating method that I can see for the class, so objects would be immutable if this method was removed.

File: radixdlt-java/src/main/java/com/radixdlt/client/atommodel/quarks/ChronoQuark.java (hunk 1 above)
Comment (talekhinezh) [reply radixdlt-java/src/main/java/com/radixdlt/client/atommodel/quarks/ChronoQuark.java]: Agreed

Comment (flotothemoon) [reply radixdlt-java/src/main/java/com/radixdlt/client/atommodel/quarks/ChronoQuark.java]: Good point

**Pull request merged**